### PR TITLE
Mssw chapter16

### PR DESCRIPTION
**Pull request by Spinodal**

### CODE REVIEW

File: mssw/MyDate/lib/MyDate.pm [hunk 1]
```diff
@@ -0,0 +1,159 @@
+package MyDate;
+
+use 5.006;
+use strict;
+use warnings;
+
+use Carp qw( croak );
+use Data::Dumper;
```
Comment (alexdeas) [inline mssw/MyDate/lib/MyDate.pm:8]: Dumped

Comment (SimonGross) [inline mssw/MyDate/lib/MyDate.pm:8]: :-1:

Comment (utopia-2) [inline mssw/MyDate/lib/MyDate.pm:8]: :disappointed:

Comment (thruddle) [inline mssw/MyDate/lib/MyDate.pm:8]: and you were doing so well last week...

Comment (utopia-2) [inline mssw/MyDate/lib/MyDate.pm:8]: I am, and I cannot understate this, SERIOUSLY CHEEZED about this. :anguished: :feelsgood:

Comment (thruddle) [inline mssw/MyDate/lib/MyDate.pm:8]: I'm guessing the lack of a cheese emoji is just making things worse?

Comment (SimonGross) [inline mssw/MyDate/lib/MyDate.pm:8]: sad :mouse:

File: mssw/MyDate/lib/MyDate.pm [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    my %properties = (
+        date  => 'mday',
+        month => 'fullmonth',
+        year  => 'year',
+    );
+
+    our $AUTOLOAD;
+
+    my ($method) = ( $AUTOLOAD =~ /::(\w+)$/ );
+    if ( grep $method eq $_, keys %properties ) {
```
Comment (utopia-2) [inline mssw/MyDate/lib/MyDate.pm:56]: It'd be easier to just do `exists $properties{$method}`, avoiding `O(N)` lookup is what hashtables are for.

File: mssw/MyDate/lib/MyDate.pm [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+for all your autoload-y needs
+
+=cut
+
+sub AUTOLOAD {
+    my %properties = (
+        date  => 'mday',
+        month => 'fullmonth',
+        year  => 'year',
+    );
```
Comment (utopia-2) [inline mssw/MyDate/lib/MyDate.pm:51]: This'll construct this small hash each time.

File: mssw/MyDate/lib/MyDate.pm [hunk 4]
```diff
@@ -0,0 +1,159 @@
+package MyDate;
+
+use 5.006;
+use strict;
+use warnings;
+
+use Carp qw( croak );
```
Comment (dickyince) [inline mssw/MyDate/lib/MyDate.pm:7]: Pre dumper dump

**Pull request merged**